### PR TITLE
contract to split ETH payments from contract wallet

### PR DESCRIPTION
**Pull request by johannbarbie**
should split an ETH payment to array of recipients according to their shares.

### CODE REVIEW

File: contracts/misc/PaymentSplitter.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+pragma solidity 0.5.2;
+
+contract PaymentSplitter {
+
+ /**
+  * Transfers given token to multiple recipients as specified by _recepients and _splits arrays
+  *
+  * @dev This contract should have enough allowance of _tokenAddr from _payerAddr
+  * @param _recipients Array of payment recipients
+  * @param _splits Array of amounts for _tokenAddr ERC20 to transfer to corresponding recipient.
```
Comment (troggy) [inline contracts/misc/PaymentSplitter.sol:16]:
```suggestion
  * @param _splits Array of amounts in wei to transfer to corresponding recipient.
```

File: contracts/misc/PaymentSplitter.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+pragma solidity 0.5.2;
+
+contract PaymentSplitter {
+
+ /**
+  * Transfers given token to multiple recipients as specified by _recepients and _splits arrays
+  *
+  * @dev This contract should have enough allowance of _tokenAddr from _payerAddr
```
Comment (troggy) [inline contracts/misc/PaymentSplitter.sol:14]:
```suggestion
```

File: contracts/misc/PaymentSplitter.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * This source code is licensed under the Mozilla Public License, version 2,
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+pragma solidity 0.5.2;
+
+contract PaymentSplitter {
+
+ /**
+  * Transfers given token to multiple recipients as specified by _recepients and _splits arrays
```
Comment (troggy) [inline contracts/misc/PaymentSplitter.sol:12]:
```suggestion
  * Transfers ether to multiple recipients as specified by _recepients and _splits arrays
```

File: contracts/misc/PaymentSplitter.sol (hunk 1 above)
Comment (troggy) [inline contracts/misc/PaymentSplitter.sol:16]:
```suggestion
  * @param _splits Array of amounts in wei to transfer to corresponding recipient.
```

File: test/paymentSplitter.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  beforeEach(async () => {
+    paymentSplitter = await PaymentSplitter.new();
+  });
+
+  it('is splitable', async () => {
+    const balanceBefore = await web3.eth.getBalance(accounts[1]);
+    await paymentSplitter.split([accounts[1], accounts[2]], [500, 500], {value: '100000000'});
+    const balanceAfter = await web3.eth.getBalance(accounts[1]);
+    assert(balanceAfter !== balanceBefore, "split not performed");
```
Comment (troggy) [inline test/paymentSplitter.js:22]: should we check the actually value transferred in tests?

File: test/paymentSplitter.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  it('needs no splitting if 1 receiver', async () => {
+    const balanceBefore = await web3.eth.getBalance(accounts[1]);
+    await paymentSplitter.split([accounts[1]], [1], {value: '100000000'});
+    const balanceAfter = await web3.eth.getBalance(accounts[1]);
+    assert(balanceAfter !== balanceBefore, "split not performed");
+  });
+
+  it('needs no splitting if no receiver', async () => {
```
Comment (troggy) [inline test/paymentSplitter.js:32]:
```suggestion
  it('needs no splitting if no value to split', async () => {
```

File: test/paymentSplitter.js (hunk 4 above)
Comment (johannbarbie) [reply test/paymentSplitter.js:22]: yeah, i didn't know how, will add.

File: contracts/misc/PaymentSplitter.sol [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    uint256 amount = msg.value;
+    require(_recipients.length == _splits.length, "splits and recipients should be of the same length");
+
+    uint256 sumShares = 0;
+    for (uint i = 0; i < _recipients.length; i++) {
+      sumShares += _splits[i];
+    }
+
+    for (uint i = 0; i < _recipients.length - 1; i++) {
+      _recipients[i].transfer(amount * _splits[i] / sumShares);
```
Comment (troggy) [inline contracts/misc/PaymentSplitter.sol:30]: `amount * _splits[i]` can overflow here. Should we use SafeMath?

Comment (johannbarbie) [reply contracts/misc/PaymentSplitter.sol:30]: yeah, very good catch

**Pull request merged**